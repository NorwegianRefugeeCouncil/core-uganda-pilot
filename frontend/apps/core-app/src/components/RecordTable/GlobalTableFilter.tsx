import React from 'react';
import { useAsyncDebounce } from 'react-table';
import { FormControl, Input } from 'native-base';
import { NativeSyntheticEvent, TextInputChangeEventData } from 'react-native';

import { Input as InputWeb } from '../Web/Input';

import { RecordTableEntry, SortedFilteredTable } from './types';

type Props<T> = {
  table: SortedFilteredTable<T>;
};

export const GlobalTableFilter: React.FC<Props<RecordTableEntry>> = ({
  table,
}) => {
  const {
    state: { globalFilter },
    setGlobalFilter,
  } = table;

  const [value, setValue] = React.useState(globalFilter);
  const onChange = useAsyncDebounce((v) => {
    setGlobalFilter(v || undefined);
    setValue(v);
  });

  return (
    <FormControl>
      <FormControl.Label>Beneficiary Name</FormControl.Label>
      {/* <Input */}
      {/*  type="text" */}
      {/*  placeholder="Search" */}
      {/*  value={value || ''} */}
      {/*  onChange={onChange} */}
      {/* /> */}
      <InputWeb
        type="text"
        placeholder="Search"
        value={value || ''}
        onChange={onChange}
      />
    </FormControl>
  );
};
