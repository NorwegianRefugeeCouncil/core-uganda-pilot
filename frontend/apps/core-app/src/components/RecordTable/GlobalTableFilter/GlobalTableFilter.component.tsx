import React from 'react';
import { FormControl, Input } from 'native-base';

type Props = {
  handleChange: (v: string) => void;
  value: string;
};

export const GlobalTableFilterComponent: React.FC<Props> = ({
  handleChange,
  value,
}) => {
  return (
    <FormControl>
      <FormControl.Label>Beneficiary Name</FormControl.Label>
      <Input
        type="text"
        placeholder="Search"
        value={value || ''}
        onChangeText={handleChange}
      />
    </FormControl>
  );
};
