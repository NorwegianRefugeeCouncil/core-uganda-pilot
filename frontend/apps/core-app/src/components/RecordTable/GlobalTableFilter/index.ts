export { GlobalTableFilterContainer as GlobalTableFilter } from './GlobalTableFilter.container';
