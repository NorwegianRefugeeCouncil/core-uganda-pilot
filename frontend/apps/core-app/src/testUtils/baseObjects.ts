import { FormDefinition, FormType, Record } from 'core-api-client';

export const baseForm: FormDefinition = {
  id: 'form1',
  code: '',
  name: 'form 1',
  databaseId: 'databaseId1',
  folderId: 'folderId1',
  formType: FormType.DefaultFormType,
  fields: [],
};

export const baseTextField = {
  id: 'field1',
  name: 'field 1',
  description: 'description 1',
  code: '',
  required: false,
  key: false,
  fieldType: {
    text: {},
  },
};

export const baseMultilineTextField = {
  id: 'field2',
  name: 'field 2',
  description: 'description 2',
  code: '',
  required: true,
  key: true,
  fieldType: {
    multilineText: {},
  },
};

export const baseRecord1: Record = {
  id: 'record1',
  values: [
    {
      value: 'text',
      fieldId: 'field1',
    },
    {
      value: 'multi line text',
      fieldId: 'field2',
    },
  ],
  formId: 'form1',
  databaseId: 'databaseId1',
  ownerId: undefined,
};
