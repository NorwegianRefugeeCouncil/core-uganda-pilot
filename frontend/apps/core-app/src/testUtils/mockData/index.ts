export { makeRecord } from './Record';
export { makeForm } from './Form';
export { makeField } from './FormField';
