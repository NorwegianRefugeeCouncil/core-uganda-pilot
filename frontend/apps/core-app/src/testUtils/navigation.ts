export const mockNavigationProp = {
  dispatch: jest.fn(),
  navigate: jest.fn(),
  reset: jest.fn(),
  goBack: jest.fn(),
  isFocused: jest.fn(),
  canGoBack: jest.fn(),
  getParent: jest.fn(),
  getState: jest.fn(),
  setParams: jest.fn(),
  setOptions: jest.fn(),
  addListener: jest.fn(),
  removeListener: jest.fn(),
  replace: jest.fn(),
  push: jest.fn(),
  pop: jest.fn(),
  popToTop: jest.fn(),
};
