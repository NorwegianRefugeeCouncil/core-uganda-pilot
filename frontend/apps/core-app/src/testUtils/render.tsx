import * as React from 'react';
import { NavigationContainer } from '@react-navigation/native';
import {
  render as r,
  RenderAPI,
  RenderOptions,
} from '@testing-library/react-native';
import { render as rWeb } from '@testing-library/react';
import { NativeBaseProvider } from 'native-base';
import { theme } from 'core-design-system';

const inset = {
  frame: { x: 0, y: 0, width: 0, height: 0 },
  insets: { top: 0, left: 0, right: 0, bottom: 0 },
};

export const render = (
  component: React.ReactElement<any>,
  options?: RenderOptions | undefined,
): RenderAPI =>
  r(
    <NativeBaseProvider theme={theme} initialWindowMetrics={inset}>
      <NavigationContainer>{component}</NavigationContainer>
    </NativeBaseProvider>,
    options,
  );

export const renderWeb = (
  component: React.ReactElement<any>,
  options?: RenderOptions | undefined,
) =>
  rWeb(
    <NativeBaseProvider theme={theme} initialWindowMetrics={inset}>
      <NavigationContainer>{component}</NavigationContainer>
    </NativeBaseProvider>,
    options,
  );
