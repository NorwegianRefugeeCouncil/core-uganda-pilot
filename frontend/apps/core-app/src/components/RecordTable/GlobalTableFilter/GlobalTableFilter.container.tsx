import React from 'react';
import { FilterValue, useAsyncDebounce } from 'react-table';

import { RecordTableEntry, SortedFilteredTable } from '../types';

import { GlobalTableFilterComponent } from './GlobalTableFilter.component';

type Props<T> = {
  table: SortedFilteredTable<T>;
  globalFilter?: FilterValue;
  setGlobalFilter?: (filterValue: FilterValue) => void;
};

export const GlobalTableFilterContainer: React.FC<Props<RecordTableEntry>> = ({
  table,
}) => {
  const {
    state: { globalFilter },
    setGlobalFilter,
  } = table;

  const [value, setValue] = React.useState(globalFilter);

  const handleChange = useAsyncDebounce((v) => {
    setGlobalFilter(v || undefined);
    setValue(v);
  }, 20);

  return (
    <GlobalTableFilterComponent handleChange={handleChange} value={value} />
  );
};
