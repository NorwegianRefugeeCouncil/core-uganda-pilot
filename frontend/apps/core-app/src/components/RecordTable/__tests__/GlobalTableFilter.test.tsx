/**
 * @jest-environment jsdom
 */

import React from 'react';
import { fireEvent, waitFor } from '@testing-library/react';

import { render, renderWeb } from '../../../testUtils/render';
import { GlobalTableFilter } from '../GlobalTableFilter';

jest.mock('react-native/Libraries/Utilities/Platform', () => ({
  OS: 'web',
  select: () => null,
}));

describe('GlobalTableFilter', () => {
  it('should match the snapshot', () => {
    const { toJSON } = render(
      <GlobalTableFilter table={{ state: {}, preGlobalFilteredRows: [] }} />,
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('should call onChange handler', async () => {
    const setGlobalFilterSpy = jest.fn();

    const { getByPlaceholderText, debug } = renderWeb(
      <GlobalTableFilter
        table={{
          state: {},
          preGlobalFilteredRows: [],
          setGlobalFilter: setGlobalFilterSpy,
        }}
      />,
    );
    debug();

    const input = getByPlaceholderText('Search');
    fireEvent.input(input, 'newFilter');

    await waitFor(() =>
      expect(setGlobalFilterSpy).toHaveBeenCalledWith('newFilter'),
    );
  });
});
