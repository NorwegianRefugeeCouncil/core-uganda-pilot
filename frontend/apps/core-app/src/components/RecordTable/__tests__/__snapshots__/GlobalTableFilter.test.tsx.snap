// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GlobalTableFilter should match the snapshot 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    dataSet={Object {}}
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <View
      accessibilityRole="label"
      dataSet={Object {}}
      feedbackId="field-react-aria-1-feedback"
      hasFeedbackText={false}
      hasHelpText={false}
      helpTextId="field-react-aria-1-helptext"
      isDisabled={false}
      isInvalid={false}
      isReadOnly={false}
      isRequired={false}
      labelId="field-react-aria-1-label"
      nativeID="field-react-aria-1-label"
      setHasFeedbackText={[Function]}
      setHasHelpText={[Function]}
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "flex-start",
          "marginBottom": 2,
          "marginRight": 12,
        }
      }
    >
      <Text
        dataSet={Object {}}
        level="1"
        style={
          Object {
            "backgroundColor": undefined,
            "color": "#666666",
            "fontFamily": "Roboto_500Medium",
            "fontSize": 18,
            "fontStyle": undefined,
            "fontWeight": undefined,
            "letterSpacing": 0,
            "lineHeight": 26,
            "textAlign": "auto",
            "textDecorationLine": undefined,
          }
        }
      >
        Beneficiary Name
      </Text>
    </View>
    <View
      accessible={true}
      dataSet={Object {}}
      editable={true}
      isHovered={false}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholderTextColor="#a3a3a3"
      secureTextEntry={false}
      style={
        Array [
          Object {},
          Array [
            Object {
              "backgroundColor": "#FFFFFF",
              "borderColor": "#666666",
              "borderRadius": 3,
              "borderStyle": "solid",
              "borderWidth": 1,
              "color": "#000000",
              "fontFamily": "Roboto_400Regular",
              "fontSize": 18,
              "fontStyle": undefined,
              "fontWeight": undefined,
              "lineHeight": 26,
              "overflow": "visible",
              "paddingBottom": 12,
              "paddingLeft": 12,
              "paddingRight": 12,
              "paddingTop": 12,
              "width": undefined,
            },
            Object {},
          ],
        ]
      }
      value=""
    >
      <input
        className="sc-bdvvtL iLuNrv"
        onChange={[Function]}
        placeholder="Search"
        role="search"
        type="text"
        value=""
      />
    </View>
  </View>
</RNCSafeAreaProvider>
`;
